pdns/dnsdist-lua2.cc

```diff
@@ -769,6 +769,47 @@ void moreLua(bool client)
 #endif
       });
 
+     g_lua.registerFunction<void(DNSDistProtoBufMessage::*)(std::string)>("setTag", [](DNSDistProtoBufMessage& message, const std::string& strValue) {
+
+      message.addTag(strValue);
+     });
+
+     g_lua.registerFunction<void(DNSDistProtoBufMessage::*)(vector<pair<int, string>>)>("setTagArray", [](DNSDistProtoBufMessage& message, const vector<pair<int, string>>&tags) {
+
+
+      for (const auto& tag : tags)
+        {
+          message.addTag(tag.second);
+        }
+     });
+
+     g_lua.registerFunction<void(DNSDistProtoBufMessage::*)(boost::optional <time_t> sec, boost::optional <uint> uSec)>("setProtobufResponseType",
+                                        [](DNSDistProtoBufMessage& message, boost::optional <time_t> sec, boost::optional <uint> uSec) {
+
+        message.setType(DNSProtoBufMessage::Response);
+
+        message.setQueryTime(sec?*sec:0, uSec?*uSec:0);
+
+     });
+
+     g_lua.registerFunction<void(DNSDistProtoBufMessage::*)(const std::string&, uint uType, uint uClass, uint uTTL, vector<pair<int, int>> )>("setProtobufResponseRR", [](DNSDistProtoBufMessage& message,
```

Comment:
Please use `uint16_t` for type and class, `uint32_t` for ttl. A simple `std:string` might be more suited for `blobData`, it would prevent to have to allocate an array and copy the content to it?